### DISCUSSION TITLE
Log Infinity

### DISCUSSION THREAD

**Discussion by RanggaGemilang** [teamalgoritma/community#323] [Q&A] [answered]
Halo saya ingin menggunakan data log-scaling untuk pemodelan time-series saya. Namun dikarenakan pada data aslinya terdapat nilai 0, data tersebut berubah menjadi -Inf. Sehingga data log-scaling saya tidak bisa digunakan untuk pemodelan.

![log infinity2](https://user-images.githubusercontent.com/107049922/191590845-f9f51f29-eb3d-487f-8ec0-99cd660be9a5.png)

Berikut adalah contoh datanya:

![Log infinity](https://user-images.githubusercontent.com/107049922/191590918-34e59d37-6995-4122-8715-2c276b865b65.png)

Apakah ada workaround yang bisa saya lakukan, atau ada referensi lain untuk melakukan scaling data untuk time-series?

Terima kasih sebelumnya :)

**Comment by NabiilahArdini** [comment, marked as answer]
Halo Bapak @RanggaGemilang ,

Apakah Bapak sudah coba tanpa penggunaan transformasi log? Bila hasil log terdapat infinite maka data memang tidak cocok untuk di transformasikan log. Mungkin Bapak bisa menggunakan scaling biasa dengan fungsi `scale()`. Kemudian hasil prediksinya dikembalikan lagi ke nilai aslinya dengan proses unscale. Untuk unscale data dapat melihat diskusi berikut: https://github.com/teamalgoritma/community/discussions/306

**Reply by RanggaGemilang** [reply]
Halo mba @NabiilahArdini , untuk tanpa transformasi log memang bisa mba, namun saya ingin mencoba scaling agar saya bisa melakukan tes lagi: apakah prediksi saya bisa lolos uji asumsi normalitas atau tidak. (Karena di prediksi awal dengan **data tanpa scaling**, residual dari prediksi saya **tidak lolos uji asumsi normalitas**.)

Merunut dari diskusi yang dilakukan disini (https://stats.stackexchange.com/questions/96598/what-if-the-residual-series-from-arima-not-looks-normal), 

Bahwa salah satu cara untuk mengatasi hal tersebut adalah **melakukan log-transformation** pada data kita. Maka dari itu saya mencoba untuk melakukan hal yang sama dalam kasus ini. Namun memang karena ternyata datanya tidak bisa, saya akan mencoba cara yang diberikan mba Nabilah diatas. Jika saya ada cara pemikiran yang salah mengenai case ini, mohon untuk informasinya yaa mba Nabiilah.

Terima kasih :)
